Feature/2017 12 20 gauge blog

MartijnW49 opened the pull request.

MartijnW49 commented:
Thanks @MrJean ! If ok, I'd like to wait untill friday to release it, since Tim will be releasing his post tomorrow and this one is kind of related.

MrJean commented:
@MartijnW49 fine by me! Make sure to update your date to reflect the publishing date :)

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+That's exactly why you should have automated tests do it for you!
+
+[Gauge](https://getgauge.io/){:target="_blank"} is yet another test automation tool that serves that purpose.
+The founder is ThoughtWorks, the company that also created  <a href="http://www.seleniumhq.org/" target="_blank">Selenium</a> and <a href="https://www.gocd.org/" target="_blank">GoCD</a>.
```
You could make ThoughtWorks link to https://www.thoughtworks.com.

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+Others may be supported by the community like JavaScript.
+You can define steps written in either one of those languages.
+These are the actual tests and can be identified by a sentence you can choose yourself.
+By combining these sentences, you can write test scenarios which are written in MarkDown.
```
Markdown (lowercase 'd')

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+To start a new project, create a new folder and run `gauge init java` in it.
+This will setup a basic Gauge project.
+Then we need to add a [pom.xml](https://github.com/getgauge-examples/java-maven-selenium/blob/master/pom.xml){:target="_blank"} file ourselves, because we need some dependencies such as Selenium WebDriver.
+Next, WebDriver needs to be setup in our test code.
```
set up

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+</p>
+
+### Writing the specs
+The specs are written in MarkDown.
```
Markdown

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+The specs are written in MarkDown.
+Each spec file starts with a title and is underlined with `===`.
+Next, some steps can be defined that will be run before each scenario.
+When listing steps, you need to prefix each step with an asterix (*) as in a MarkDown list.
```
Markdown

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+When listing steps, you need to prefix each step with an asterix (*) as in a MarkDown list.
+After that, the actual scenarios can be written.
+They start with a title and are underlined with `---`.
+Again, the steps for a scenario should be listed as in a MarkDown list.
```
Markdown

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+### Concepts
+If you find yourself repeating the same sequence of steps over and over,
+you could combine those steps into one step using concepts.
+These are also written in MarkDown and you can pass arguments the same way as in the Java `@Step` annotations.
```
Markdown

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+
+`mvn gauge:execute -DspecsDir=specs -DinParallel=true`
+
+Running this command will look how many CPU cores your computer has and start up that many streams.
```
I think it sounds a bit better like this:
Running this command will start up a stream for each CPU core your computer has.

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+`mvn gauge:execute -DspecsDir=specs -DinParallel=true -Dnodes=2`
+
+Now, some specs may take longer to run than others.
+By default specs are divided dynamically over the streams.
```
By default, specs

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+It's possible to change this so the specs are divided on startup, but the command is deprecated and will be removed.
+
+### Making tests independent
+To make use of this parallel execution, you'll have to make sure your tests don't rely on each other.
```
make sure **that** your

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+In the [example by Gauge](https://github.com/getgauge-examples/java-maven-selenium){:target="_blank"}, they have an `APP_URL` parameter in that last file.
+I recommend using that approach as well, you can get parameters in your Java code using `System.getenv("APP_URL")`.
+
+It's possible to create different environments by simply creating a new folder and add the configuration files that you want to change there.
```
... and **adding** the ...
Not sure if I understand the last part of the sentence, "the configuration files that you want to change"?
You might mean specifying config files for multiple environment if certain properties differ depending on the environment?

yannickdeturck commented on _posts/2018-01-03-Gauge-automated-testing.md:
```diff
@@ -0,0 +1,286 @@
+Create a folder named `firefox` for example and add a file called `browser.properties`.
+In that file you add `browser = FIREFOX`.
+When you then run the tests with the `firefox` environment, it will use FireFox as a browser instead.
+(This only works if you have your project setup like in the example, the [Driver](https://github.com/getgauge-examples/java-maven-selenium/tree/master/src/test/java/utils/driver){:target="_blank"} and [DriverFactory](https://github.com/getgauge-examples/java-maven-selenium/tree/master/src/test/java/utils/driver){:target="_blank"} files are required here.)
```
set up

MartijnW49 commented:
Thanks @yannickdeturck ! I fixed the nitpickings and clarified the part about the configuration. I also updated the date to tomorrow, so if you agree with the changes, you may merge this PR :)
